### PR TITLE
fixing up missing static file issue on swagger-ui

### PR DESCRIPTION
**Pull request by kozmaz87**
I added missing static files for swagger ui. There is still a tiny issue that the link under the description now says: /swagger/ and that is wrong... not sure how to fix that one....

### CODE REVIEW

**Comment by sloria** [comment]
I actually haven't tried the automatic swagger-ui feature for this project. @jmcarp How is this expected to work? Are users supposed to install the assets manually after installing `flask-apispec` themselves?

**Comment by kozmaz87** [comment]
There is `https://github.com/jmcarp/flask-apispec/blob/master/tasks.py` file in which there is an install section where it goes to npm to get the latest swagger-ui but it does not end up being in the python package nor is it user replaceable because the static folder is being looked for in the site-packages/flask-apispec...etc path in which the user should not manually modify anything. I basically changed this dynamic approach and statically moved in the swagger ui, not to mention that the templating requires all the `src` attributes to be prefixed with the flask static file template handles so that it can work reliably. 

Also the whole APISPEC_SWAGGER_URL and APISPEC_SWAGGER_UI_URL distinction is calling for trouble. assuming you have a reverse proxy that maps your endopoints to some prefix or does rewrites... I am not sure if swagger ui would be able to find swagger in every possible case. Because unfortunately the javascript on the client side needs to initiate the query with the full URL otherwise it would not load. However if swagger-ui and swagger are always next to each other the logic is simple to generate the correct url and that way the config does not have to contain the full SERVER_URL, which improves portability and flexibility.

**Comment by rth** [comment]
@kozmaz87 Thanks for the PR. Would this address #28 , #57 ? cc @ArthurPBressan

**Comment by ArthurPBressan** [comment]
I'm going to check this out over the weekend, but it should solve my issue.

**Comment by ArthurPBressan** [comment]
This worked for me :+1:

File: flask_apispec/templates/swagger-ui.html [hunk 1]
```diff
@@ -1,56 +1,93 @@
...
-        },
-        docExpansion: 'none',
-        apisSorter: 'alpha',
-        showRequestHeaders: false
-      });
-
-      window.swaggerUi.load();
-    });
-  </script>
+  <link href="https://fonts.googleapis.com/css?family=Open+Sans:400,700|Source+Code+Pro:300,600|Titillium+Web:400,600,700" rel="stylesheet">
```
Comment (antonech) [inline flask_apispec/templates/swagger-ui.html]: Can it be local link?
Not always resource will be connected to internet.

**Comment by tiangolo** [comment]
@sloria or @jmcarp what do you need to check / approve this PR?

It seems to be working and fixing #28 and #57 . I'm pretty sure I was having those issues too.

Also, this PR updates the Swagger UI version.

It might not seem intuitive to add to git a compiled version of the front end code. But in this case I think it's a good idea, as it would prevent publishing to PyPi without including the front end code.

---

Also, the current "error", the fact that the Swagger UI is missing, might be due to having forgotten to generate the static files before publishing to PyPi: https://github.com/jmcarp/flask-apispec/issues/22#issuecomment-315150500

I understand that [it is needed to run](https://flask-apispec.readthedocs.io/en/latest/install.html#from-source):

```bash
pip install invoke
invoke install
```

before publishing to PyPi, for it to download and install all the JS files.

**Comment by rangeltorrezan** [comment]
Anyone have prevision when this error will solve?

**Comment by ffleandro** [comment]
Still no prevision when will this PR me merged into master?

**Comment by sloria** [comment]
Sorry for the delay on this. I'll try to look at this this weekend. Thanks for your patience.

**Comment by ffleandro** [comment]
Thanks @sloria.
I've merged this PR on my local master branch and everything works as expected.

Also, from my opinion, I think there should be a configuration for the user to specify the swagger-ui static and templates location.

I think it's a very common scenario to customize the swagger UI, for instance, to add your own branding or to adequate to a broader HTML structure.

This can easily be done with the following modifications to the file `flask_apispec/extensions.py`:

**From:**
```
blueprint = flask.Blueprint(
    'flask-apispec',
    __name__,
    static_folder='./static',
    template_folder='./templates',
    static_url_path='/flask-apispec/static',
)
```

**To:**
```
swagger_path = self.app.config.get('APISPEC_SWAGGER_UI_PATH', '.')

blueprint = flask.Blueprint(
    'flask-apispec',
     __name__,
    static_folder=swagger_path + '/static',
    template_folder=swagger_path + '/templates',
    static_url_path='/flask-apispec/static',
)
```


And then on my application specific Config file I add the following property:
`APISPEC_SWAGGER_UI_PATH = os.path.abspath(os.path.dirname(__file__)) + '/static/swagger-ui'`

Let me know if you want me to create a new PR, but I think this one needs to be merged before.

**Comment by daenney** [comment]
Is there any progress on this? It makes the package only half-useful at the moment.

**Comment by sloria** [comment]
I've fixed the automatic deployment so that the swagger-ui assets get included in the PyPI distributions. `flask-apispec==0.4.1.post0` should be working now.

Thanks anyway for the PR @kozmaz87 .

@ffleandro Feel free to send a PR with your proposed modifications.

**Pull request closed without merging**